Modules/getpath.c

```diff
@@ -1291,21 +1291,26 @@ calculate_read_pyenv(PyCalculatePath *calculate)
 }
 
 
+#define _PPCAT_NX(A, B) A ## B
+#define _PPCAT(A, B) _PPCAT_NX(A, B)
+#define _STRINGIZE_NX(A) #A
+#define _STRINGIZE(A) _STRINGIZE_NX(A)
+
 static PyStatus
 calculate_zip_path(PyCalculatePath *calculate)
 {
     PyStatus res;
 
     /* Path: <PLATLIBDIR> / "python00.zip" */
-    wchar_t *path = joinpath2(calculate->platlibdir_macro, L"python000.zip");
+    wchar_t *path = joinpath2(calculate->platlibdir_macro, L"python" _STRINGIZE(_PPCAT(PY_MAJOR_VERSION, PY_MINOR_VERSION)) L".zip");
```

Comment:
I'm not comfortable with playing with strings in the preprocessor, nor with concatenating wide strings and bytes strings: `L"python" (wchar_t*) "310" (char*) L".zip" (wchar_t*)`.

So I proposed PR #20225 with more regular (and more boring) C code.

Reply:
> concatenating wide strings and bytes strings

The C99 standard says that is defined behaviour, so we should be safe.